lib/AST/GraphOperationInfo.cpp

```diff
@@ -148,3 +129,72 @@ std::string GraphOperationInfo::getStringAttr(unsigned attrIdx,
   auto attrValue = attr.value;
   return attrValue.getStringValue().str();
 }
+
+GraphOperationInfo::OperandMarker::OperandMarker(
+    StringRef MangledName, const GraphOperationInfo &info)
+    : MangledName(MangledName) {
+  info.assertWithDump(MangledName.size() >= 2, "marker too short");
+  info.assertWithDump(MangledName.find_last_of(",") == 0, "incorrect comma");
+
+  switch (MangledName[1]) {
+  case 's':
+    Kind = OMK_Scalar;
+    assert(getName().empty());
+    break;
+  case 'i':
+    Kind = OMK_Normal;
+    break;
+  case 'L':
+    Kind = OMK_InputList;
+    break;
+  case 'e':
+    Kind = OMK_InputListElt;
+    assert(getName().empty());
+    break;
+  default:
+    info.assertWithDump(false, "unknown marker kind");
+  }
+}
+
+GraphOperationInfo::OperandMarkerKind
```

Comment:
Move getKind, getName, and getMangledName into the class declaration itself?

Reply:
moved